packages/main/src/plugin/util/port.ts

```diff
@@ -21,19 +21,19 @@ import * as net from 'net';
 /**
  * Find a free port starting from the given port
  */
-export function getFreePort(port = 0): Promise<number> {
+export async function getFreePort(port = 0): Promise<number> {
   if (port < 1024) {
     port = 9000;
   }
-  const server = net.createServer();
-  return new Promise((resolve, reject) =>
-    server
-      .on('error', (error: NodeJS.ErrnoException) =>
-        error.code === 'EADDRINUSE' ? server.listen(++port) : reject(error),
-      )
-      .on('listening', () => server.close(() => resolve(port)))
-      .listen(port),
-  );
+  let isFree = false;
+  while (!isFree) {
```

Comment:
I would add an upper limit, in case detection fails every time for some system failure, to avoid looping infinitely

Reply:
What do you mean by upper limit? 65353? or like try 100 times and stop?

Reply:
A port upper limit, like 65353

Reply:
In that case it already fails bc when it goes over the upper range, the promise in `isFreePort` rejects. 
So it will not run infinitely. It just resolve to false if the error code is `EADDRINUSE`

Reply:
I can return the starting port if that happens. Updating

Reply:
Thinking again, the function should return a free port so if that's not possible it makes sense it returns an error

Reply:
added a couple of tests to verify that it fails with a big port number